File: lang/php/src/vac.fun/algorithms/search/BinarySearch.php
<?php
    namespace vac\fun\algorithms\search\BinarySearch;

    function find(array &$arr, $key): int {
        $lo = 0;
        $hi = count($arr);

        $mid = 0;

        while ($lo < $hi) {
            $mid = $lo + (($hi - $lo) >> 1);
            switch (true) {
            case $key < $arr[$mid]:
                $hi = $mid;
                break;
            case $key > $arr[$mid]:
                $lo = $mid + 1;
                break;
            default:
                return $mid;
            }
        }

        return count($arr);
    }
?>

File: go/muse/algorithms/search/binary_search/binary_search.go
package binary_search

func Find(arr []int, key int) int {
	lo := 0
	hi := len(arr)

	for lo < hi {
		mid := lo + ((hi - lo) >> 1)
		switch {
		case key < arr[mid]:
			hi = mid
		case key > arr[mid]:
			lo = mid + 1
		default:
			return mid
		}
	}

	return len(arr)
}
